library/bigip_software_facts.py

```diff
@@ -0,0 +1,306 @@
+
+        if 'image' in self.include:
+            output['images'] = images.get_images_facts()
+        if 'hotfix' in self.include:
+            output['hotfixes'] = hotfixes.get_hotfixes_facts()
+        if 'volume' in self.include:
+            output['volumes'] = volumes.get_volumes_facts()
+
+        return output
+
+    def _filter_and_format_facts(self, fact):
+        filtered = dict()
+        listing = fact.attrs
+        for k, v in listing.items():
+            if k in Parameters.returnables:
+                filtered[str(k)] = str(v)
+        return filtered
+
+    def collection_parser(self, collection):
+        output = list()
+        if self.filter is None:
+            for item in collection:
+                output.append(self._filter_and_format_facts(item))
+        else:
+            bools = ['true', 'True', 'false', 'False']
+            key, value = self.filter
+            if value in bools:
+                value = bool(value)
+            for item in collection:
+                if hasattr(item, key):
+                    if getattr(item, key) == value:
+                        output.append(self._filter_and_format_facts(item))
+        return output
+
+
+class ImageFactManager(FactManagerBase):
+
+    def get_images_facts(self):
+        collection = self.list_images_on_device()
+        to_return = self.collection_parser(collection)
+        return to_return
+
+    def list_images_on_device(self):
+        images = self.client.api.tm.sys.software.images.get_collection()
+        return images
+
+
+class HotfixFactManager(FactManagerBase):
+
+    def get_hotfixes_facts(self):
+        collection = self.list_hotfixes_on_device()
+        to_return = self.collection_parser(collection)
+        return to_return
+
+    def list_hotfixes_on_device(self):
+        hotfixes = self.client.api.tm.sys.software.hotfix_s.get_collection()
+        return hotfixes
+
+
+class VolumeFactManager(FactManagerBase):
+
+    def get_volumes_facts(self):
+        collection = self.list_volumes_on_device()
+        to_return = self.collection_parser(collection)
+        return to_return
+
+    def list_volumes_on_device(self):
+        volumes = self.client.api.tm.sys.software.volumes.get_collection()
+        return volumes
+
+
+class ArgumentSpec(object):
+    def __init__(self):
+        self.supports_check_mode = True
+        self.argument_spec = dict(
+            include=dict(
+                type='list',
+                default=['all'],
+            ),
+            filter=dict()
+        )
+        self.f5_product_name = 'bigip'
+
+
+def main():
+    if not HAS_F5SDK:
+        raise F5ModuleError("The python f5-sdk module is required")
+
+    spec = ArgumentSpec()
+
+    client = AnsibleF5Client(
+        argument_spec=spec.argument_spec,
+        supports_check_mode=spec.supports_check_mode,
+        f5_product_name=spec.f5_product_name
+    )
+
+    try:
+        mm = FactManagerBase(client)
```

Comment:
We need to change this so that `ModuleManager` is used instead of the base manager. Only manager can be used in the main method